File: relationship-manager/analytics-service/src/main/java/com/bank/rm/analytics/controller/AnalyticsController.java
package com.bank.rm.analytics.controller;

import com.bank.rm.analytics.dto.AnalyticsDtos.DashboardMetrics;
import com.bank.rm.analytics.service.AnalyticsAggregator;
import com.bank.rm.common.dto.ApiResponse;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/analytics")
public class AnalyticsController {
    private final AnalyticsAggregator aggregator;

    public AnalyticsController(AnalyticsAggregator aggregator) {
        this.aggregator = aggregator;
    }

    @GetMapping("/dashboard")
    public ResponseEntity<ApiResponse<DashboardMetrics>> getDashboard() {
        return ResponseEntity.ok(ApiResponse.ok(aggregator.getDashboardMetrics()));
    }
}

File: relationship-manager/analytics-service/src/main/kotlin/com/bank/rm/analytics/controller/AnalyticsController.kt
package com.bank.rm.analytics.controller

import com.bank.rm.analytics.dto.*
import com.bank.rm.analytics.service.AnalyticsEventProcessor
import com.bank.rm.common.dto.ApiResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/analytics")
class AnalyticsController(private val analyticsService: AnalyticsEventProcessor) {

    @GetMapping("/dashboard")
    fun getDashboard(): ApiResponse<DashboardMetrics> =
        ApiResponse(success = true, data = analyticsService.getDashboardMetrics())

    @GetMapping("/funnel")
    fun getConversionFunnel(): ApiResponse<ConversionFunnel> =
        ApiResponse(success = true, data = analyticsService.getConversionFunnel())
}
